config/initializers/cors.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+# Be sure to restart your server when you modify this file.
+
+# Avoid CORS issues when API is called from the frontend app.
+# Handle Cross-Origin Resource Sharing (CORS) in order to accept cross-origin Ajax requests.
+
+# Read more: https://github.com/cyu/rack-cors
+
+# Rails.application.config.middleware.insert_before 0, Rack::Cors do
+#   allow do
+#     origins "example.com"
+#
+#     resource "*",
+#       headers: :any,
+#       methods: [:get, :post, :put, :patch, :delete, :options, :head]
+#   end
+# end
```

Comment:
I get this error which is given by CORS and it stops your react app from rendering the random greetings🥲.

<img width="1418" alt="image" src="https://github.com/Bleu-Yves-Sopoude/hello-rails-back-end/assets/64643995/31bdf1fc-7a87-41f4-88b6-a6f4ebd7eb9b">

To fix this, kindly add the following block of code to your `config/initializers/cors.rb file`

```

Rails.application.config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins 'http://localhost:3001' # Replace with your frontend's origin

    resource '/greeting',
      headers: :any,
      methods: [:get, :options] # Add other HTTP methods as needed
  end
end

```

After that, add `gem 'rack-cors', require: 'rack/cors'` to your `gemfile`, and run bundle install😉. this will install the rack-cors gem👍🏼.

Test your application and see if it works well😉.